crates/storage/store_db/redb.rs

```diff
@@ -767,22 +771,45 @@ impl StoreEngine for RedBStore {
             .map_err(StoreError::RLPDecode)
     }
 
-    fn set_storage_heal_paths(
-        &self,
-        accounts: Vec<(H256, Vec<Nibbles>)>,
-    ) -> Result<(), StoreError> {
-        self.write(
-            SNAP_STATE_TABLE,
-            SnapStateIndex::StorageHealPaths,
-            accounts.encode_to_vec(),
-        )
+    fn set_storage_heal_paths(&self, paths: Vec<(H256, Vec<Nibbles>)>) -> Result<(), StoreError> {
+        let key_values = paths
+            .into_iter()
+            .map(|(hash, paths)| {
+                (
+                    <H256 as Into<AccountHashRLP>>::into(hash),
+                    <Vec<Nibbles> as Into<TriePathsRLP>>::into(paths),
+                )
+            })
+            .collect();
+        self.write_batch(STORAGE_HEAL_PATHS_TABLE, key_values)
     }
 
-    fn get_storage_heal_paths(&self) -> Result<Option<Vec<(H256, Vec<Nibbles>)>>, StoreError> {
-        self.read(SNAP_STATE_TABLE, SnapStateIndex::StorageHealPaths)?
-            .map(|rlp| RLPDecode::decode(&rlp.value()))
-            .transpose()
-            .map_err(StoreError::RLPDecode)
+    fn get_storage_heal_paths(
+        &self,
+        limit: usize,
+    ) -> Result<Vec<(H256, Vec<Nibbles>)>, StoreError> {
+        // Read values
+        let txn = self.db.begin_read()?;
+        let table = txn.open_table(STORAGE_HEAL_PATHS_TABLE)?;
+        let res: Vec<(H256, Vec<Nibbles>)> = table
+            .range(<H256 as Into<AccountHashRLP>>::into(Default::default())..)?
+            .map_while(|val| {
+                val.ok()
+                    .map(|(hash, paths)| (hash.value().to(), paths.value().to()))
+            })
+            .take(limit)
+            .collect();
+        txn.close()?;
+        // Delete read values
+        let txn = self.db.begin_write()?;
+        {
+            let mut table = txn.open_table(STORAGE_HEAL_PATHS_TABLE)?;
+            for (hash, _) in res.iter() {
+                table.remove(<H256 as Into<AccountHashRLP>>::into(*hash))?;
+            }
+        }
+        txn.commit()?;
```

Comment:
Having the `get_*` method delete keys sounds confusing. Would it be too bad to split this?

Reply:
Maybe a rename? (`get_and_remove_` o `retrieve_` or something that gives the idea of consumption of elements?)

Reply:
I agree, I think `take` would be suitable here

Reply:
Updated [ba937d5](https://github.com/lambdaclass/ethrex/pull/2359/commits/ba937d596e086deeb8d0b1184f43382e43a9b661)